src/github.com/matrix-org/dendrite/syncapi/consumers/roomserver.go

```diff
@@ -96,7 +96,7 @@ func (s *Server) onMessage(msg *sarama.ConsumerMessage) error {
 		}).Panicf("roomserver output log: write event failure")
 		return nil
 	}
-	s.rp.OnNewEvent(&ev, types.StreamPosition(syncStreamPos))
+	s.notifier.OnNewEvent(&ev, types.StreamPosition(syncStreamPos))
```

Comment:
This function is called whenever there is a new event from the room server.

Previously, `RequestPool` would handle both the notification of this event and the calculation of responses. This PR changes it so `RequestPool` only needs to care about calculation of responses, NOT notifying any waiting requests.

As a result, `Server` doesn't need to know that `RequestPool` even exists, it just needs to know about the `Notifier`, which is why this has changed from being a `RequestPool` to a `Notifier`.